File: JavaScript/Algorithms/Sorting/insertion_sort.js
const numbers = [99, 44, 6, 1, 2, 5, 63, 87, 283, 4, 0];

// [1, 6, 44, 99, 2, 5, 63, 87, 283, 4, 0];

function insertionSort(arr) {
  const length = arr.length;

  for (let step = 1; step < length; step++) {
    const key = arr[step]; // 2
    let i = step - 1; // 3

    while (i >= 0 && key < arr[i]) {
      arr[i + 1] = arr[i]
      i--;
    }

    arr[i + 1] = key;
  }

  return arr;
}

console.log(insertionSort(numbers));

File: Python/Algorithms/Sorting Algorithms/insertion_sort.py
numbers = [99, 44, 6, 1, 2, 5, 63, 87, 283, 4, 0]


def insertion_sort(arr):
    length = len(arr)

    for step in range(1, length):
        key = arr[step]
        i = step - 1

        while i >= 0 and key < arr[i]:
            arr[i + 1] = arr[i]
            i -= 1

        arr[i + 1] = key

    return arr


print(insertion_sort(numbers))
